app/src/main/java/com/andrew/tasky/agenda/data/agenda/AgendaRepositoryImpl.kt

```diff
@@ -59,9 +59,8 @@ class AgendaRepositoryImpl(
             startEpochMilli = startEpochMilli,
             endEpochMilli = endEpochMilli
         ).map {
-            it.map {
-                eventEntity ->
-                eventEntity.toEvent()
+            it.map { eventEntity ->
+                eventEntity.toEvent(appContext)
```

Comment:
Every time, you're executing a suspend function in a loop (which the `map` function is), you can make a mistake that really harms the performance. Here, you're executing the mapper sequentially, that means the function will first map event1, when that's done it'll map event2, then event3, and so on.
However, mapping an event isn't dependent on mapping a previous event. So you could execute all these `toEvent()` functions in parallel rather than sequentially.
If we assume that executing `toEvent()` takes 100ms, with your approach this function would take 1000ms.
If you execute all in parallel though, the function would only take 100ms. And the larger your list gets, the bigger the performance problem gets. 

So here's how you can fix this:
```
val mappedEvents = supervisorScope {
    eventEntities.map { entity ->
        async {
            entity.toEvent(appContext)
        }
    }
    .map { it.await() }
}
```

With `supervisorScope`, you can get a reference to a coroutine scope inside a suspend function (in this case inside `getAgendaItem()`. That's needed to launch independent coroutines which run in parallel. `supervisorScope` means if one of these coroutines launched in it will fail, the others will still run. There's also `coroutineScope` which will fail all coroutines if just one of them fails. 
With `async`, we launch a new independent coroutine which holds a result. The result can then be retrieved with `await()`. Hope that's all clear, if not just ask 👍